StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/PaymentSheetVerticalViewController.swift

```diff
@@ -11,16 +11,18 @@
 @_spi(STP) import StripeUICore
 import UIKit
 
-class PaymentSheetVerticalViewController: UIViewController, FlowControllerViewController {
+class PaymentSheetVerticalViewController: UIViewController, FlowControllerViewControllerProtocol, PaymentSheetViewControllerProtocol {
     var selectedPaymentOption: PaymentSheet.PaymentOption?
     var selectedPaymentMethodType: PaymentSheet.PaymentMethodType?
-    weak var delegate: FlowControllerViewControllerDelegate?
     let loadResult: PaymentSheetLoader.LoadResult
     let configuration: PaymentSheet.Configuration
     var intent: Intent {
         return loadResult.intent
     }
     var error: Error?
+    let isFlowController: Bool
+    weak var flowControllerDelegate: FlowControllerViewControllerDelegate?
+    weak var paymentSheetDelegate: PaymentSheetViewControllerDelegate?
```

Comment:
This is a bit weird, TBD if/how we'll make this nicer.  I thought about using an enum like
````
enum Delegate {
 case fc(FlowControllerViewControllerDelegate)
 case ps(...)
}
```
but that would make the delegate strong instead of weak.

Reply:
I think the current approach seems reasonable, should we assert that we don't have both set at one time, that seems like invalid state?

Reply:
Hmm if we are worried about that invalid state I think I'd rather try to find some way to make it impossible vs. asserting. I don't think I am though, like...I don't think anything even breaks if both are set?